src/games/supported/Freeway.cpp

```diff
@@ -95,3 +98,38 @@ void FreewaySettings::loadState(Deserializer & ser) {
   m_terminal = ser.getBool();
 }
 
+// returns a list of mode that the game can be played in
+ModeVect FreewaySettings::getAvailableModes() {
+    ModeVect modes(8);
+    for(unsigned i = 0; i < 8; i++) {
+        modes[i] = i;
+    }
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void FreewaySettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment) {
+    if(m < 8) { /*m >= 0 is implicit, since m is an unsigned int*/
+        m_mode = m;
+        // read the mode we are currently in
+        unsigned char mode = readRam(&system, 0x80);
+        // press select until the correct mode is reached
+        while(mode != m_mode){
+            environment.pressSelect(1);
+            mode = readRam(&system, 0x80);
+        }
+        //reset the environment to apply changes.
+        environment.softReset();
+    } else {
+        throw std::runtime_error("This mode doesn't currently exist for this game");
+    }
+ }
+ 
+DifficultyVect FreewaySettings::getAvailableDifficulties(){
+    DifficultyVect diff;

```

Comment:
I think in C++11 you can do diff ({0, 1}).

Reply:
It seems so, however, I tried to do:

`DifficultyVect diff = {0, 1};`

and I got the following error:
` error: non-aggregate type 'DifficultyVect' (aka 'vector<unsigned int>') cannot be initialized with an initializer list`

It could be because the `CMakeLists` file is not actually using c++11? When you merged the code, is there any chance not everyone will compile with c++11, otherwise I don't know what it is.

Reply:
Very possibly missing `-std=c++11` in the g++ flags. I'm surprised the code itself compiling, given that I switched on a few other C++11 specific pieces of code.

Reply:
I added the `-std=c+11` flag to `CMakeLists.txt` and now I'm able to initialize the vector the way you want.